modules/postprocessing/src/processors/depthoffield.cpp

```diff
@@ -0,0 +1,427 @@
+        camera_.setCamera(std::unique_ptr<Camera>(ogCamera_->clone()));
+    }
+}
+
+void DepthOfField::setupRecursion(size2_t dim, int maxEvalCount, std::shared_ptr<const Image> img) {
+    ogCamera_.reset(camera_.get().clone());
+    ogType_ = camera_.get().getClassIdentifier();
+    if (ogType_ != PerspectiveCamera::classIdentifier) {
+        LogWarn("Intended for use with Perspective Camera. Unexpected behavior may follow.");
+    }
+    camera_.setCamera(SkewedPerspectiveCamera::classIdentifier);
+
+    if (approximate_) {
+        // Prepare light field
+        dimLightField_ = size3_t(dim.x, dim.y, viewCountApprox_.get() + simViewCountApprox_.get());
+        lightField_ = std::make_shared<Volume>(dimLightField_, DataFormat<vec4>::get());
+        lightFieldDepth_ = std::make_shared<Volume>(dimLightField_, DataFormat<float>::get());
+        float* lightFieldDepthData = static_cast<float*>(
+            lightFieldDepth_->getEditableRepresentation<VolumeRAM>()->getData());
+        std::fill(lightFieldDepthData,
+                  lightFieldDepthData + dimLightField_.x * dimLightField_.y * dimLightField_.z,
+                  -1.f);
+
+        // Prepare Halton sequence
+        haltonX_ = util::haltonSequence<float>(2, simViewCountApprox_.get());
+        haltonImg_ = std::make_shared<Image>(size2_t(simViewCountApprox_.get(), 1),
+                                             DataFormat<float>::get());
+        float* haltonData = static_cast<float*>(
+            haltonImg_->getColorLayer()->getEditableRepresentation<LayerRAM>()->getData());
+        std::copy(haltonX_.begin(), haltonX_.end(), haltonData);
+    } else {
+        // Prepare accumulation buffer
+        prevOutImg_ = std::make_shared<Image>(dim, DataFormat<vec4>::get());
+        nextOutImg_ = std::make_shared<Image>(dim, DataFormat<vec4>::get());
+
+        // Prepare Halton sequences
+        haltonX_ = util::haltonSequence<float>(2, maxEvalCount);
+        haltonY_ = util::haltonSequence<float>(3, maxEvalCount);
+    }
+
+    // Update focus depth and aperture ranges based on the depths in the scene, ignoring
+    // background points and including the current value.
+    const float* inputDepthData = static_cast<const float*>(
+        img->getRepresentation<ImageRAM>()->getDepthLayerRAM()->getData());
+    float minDepthNdc = 1;
+    float maxDepthNdc = 0;
+    for (const float* f = inputDepthData; f < inputDepthData + dim.x * dim.y; f++) {
+        minDepthNdc = std::min(minDepthNdc, *f);
+        if (*f < 1) {
+            maxDepthNdc = std::max(maxDepthNdc, *f);
+        }
+    }
+
+    double nearClip = camera_.getNearPlaneDist();
+    double farClip = camera_.getFarPlaneDist();
+    float minDepthWorld = nearClip * farClip / (minDepthNdc * (nearClip - farClip) + farClip);
+    float maxDepthWorld = nearClip * farClip / (maxDepthNdc * (nearClip - farClip) + farClip);
+
+    focusDepth_.setMinValue(std::min(minDepthWorld, focusDepth_.get()));
```

Comment:
Not allowed to set properties in the process function.
Also, either lock the network or use property.set with all values at the same time because otherwise the network will be evaluated multiple times